src/Build/Definition/Project.cs

```diff
@@ -1117,33 +1129,169 @@ public List<GlobResult> GetAllGlobs(string itemType)
             return GetAllGlobs(GetItemElementsByType(GetEvaluatedItemElements(), itemType));
         }
 
+        // represents cumulated remove information for a particular item type
+        private struct CumulativeRemoveElementData
+        {
+            private ImmutableList<IMSBuildGlob>.Builder _globs;
+            private ImmutableHashSet<string>.Builder _fragmentStrings;
+
+            public IEnumerable<IMSBuildGlob> Globs => _globs.ToImmutable();
```

Comment:
Sanity. Just in case someone casts it back to builder and mutates it :)

The operation itself is not that expensive, it traverses the backing tree and toggles an `_frozen` bool to true.